src/lib/caching/structures/Message.ts

```diff
@@ -254,6 +254,15 @@ export class Message extends Structure {
 		return new ReactionCollector(this, options).collect();
 	}
 
+	/**
+	 * Crosspost this message.
+	 * @since 0.0.1
+	 */
+	public async crosspost() {
+		const channel = this.channel as NewsChannel;
```

Comment:
instead of casting, you need to assert that this.channel is a instanceof NewsChannel, and reject with an error if it's not.